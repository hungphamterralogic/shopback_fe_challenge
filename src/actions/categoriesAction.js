import { createAction } from 'redux-actions';

export const initCategories = createAction('CATEGORIES_INIT');
export const selectCategory = createAction('CATEGORY_SELECT');

export default this;
