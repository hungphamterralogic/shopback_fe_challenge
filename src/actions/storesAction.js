import { createAction } from 'redux-actions';

export const initStores = createAction('STORES_INIT');
export const selectStore = createAction('STORE_SELECT');

export default this;
