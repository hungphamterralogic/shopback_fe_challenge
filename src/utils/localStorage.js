const setValueIntoKey = (key, value) => localStorage.setItem(key, value);

const getValueFromKey = key => localStorage.getItem(key);

module.exports = {
  setValueIntoKey,
  getValueFromKey
};
