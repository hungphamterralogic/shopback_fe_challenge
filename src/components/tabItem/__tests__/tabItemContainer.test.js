import React from 'react';

import { genInitialMockStore, renderWithStore } from 'test/reduxComponentTestHelpers';

import detectDevice from 'utils/detectDevice';
import * as categoriesAction from 'actions/categoriesAction';
import localStorage from 'utils/localStorage';

import TabItemContainer from '../tabItemContainer';

describe('TabItemContainer', () => {
  const props = {
    id: 'idTest',
    name: 'nameTest',
    iconUrl: 'iconUrlTest'
  };

  describe('display properly', () => {
    describe('is selected', () => {
      let mockStore;

      beforeAll(() => {
        mockStore = genInitialMockStore({
          'categories.selected': props.id
        });
      });

      test('is mobile', () => {
        jest.spyOn(detectDevice, 'isMobile').mockImplementationOnce(() => true);

        const renderer = renderWithStore(<TabItemContainer {...props} />, mockStore);
        expect(renderer.toJSON()).toMatchSnapshot();
      });

      test('is not mobile', () => {
        jest.spyOn(detectDevice, 'isMobile').mockImplementationOnce(() => false);

        const renderer = renderWithStore(<TabItemContainer {...props} />, mockStore);
        expect(renderer.toJSON()).toMatchSnapshot();
      });
    });

    describe('is not selected', () => {
      let mockStore;

      beforeAll(() => {
        mockStore = genInitialMockStore({
          'categories.selected': `differentThan${props.id}`
        });
      });

      test('is mobile', () => {
        jest.spyOn(detectDevice, 'isMobile').mockImplementationOnce(() => true);

        const renderer = renderWithStore(<TabItemContainer {...props} />, mockStore);
        expect(renderer.toJSON()).toMatchSnapshot();
      });

      test('is not mobile', () => {
        jest.spyOn(detectDevice, 'isMobile').mockImplementationOnce(() => false);

        const renderer = renderWithStore(<TabItemContainer {...props} />, mockStore);
        expect(renderer.toJSON()).toMatchSnapshot();
      });
    });
  });

  test('on click', () => {
    const selectCategoryMock = jest.spyOn(categoriesAction, 'selectCategory');
    const setValueIntoKeyMock = jest.spyOn(localStorage, 'setValueIntoKey');

    const mockStore = genInitialMockStore();
    const renderer = renderWithStore(<TabItemContainer {...props} />, mockStore);

    (function clickOnTabItem() {
      const tabItem = renderer.root.find(el => el.type === 'div');
      tabItem.props.onClick();
    })();

    expect(selectCategoryMock.mock.calls.length).toBe(1);
    expect(selectCategoryMock).toBeCalledWith(props.id);

    expect(setValueIntoKeyMock.mock.calls.length).toBe(1);
    expect(setValueIntoKeyMock).toBeCalledWith('category', props.id);
  });
});
