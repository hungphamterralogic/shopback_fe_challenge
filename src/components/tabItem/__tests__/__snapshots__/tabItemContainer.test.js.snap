// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItemContainer display properly is not selected is mobile 1`] = `
<div
  className="TabItem-tabItem-0-1-1 TabItem-tabItemMobile-0-1-3"
  onClick={[Function]}
>
  <img
    alt="icon"
    className="TabItem-icon-0-1-5"
    src="iconUrlTest"
  />
  <span
    className="TabItem-name-0-1-6"
  >
    nameTest
  </span>
</div>
`;

exports[`TabItemContainer display properly is not selected is not mobile 1`] = `
<div
  className="TabItem-tabItem-0-1-1 TabItem-tabItemDestop-0-1-2"
  onClick={[Function]}
>
  <img
    alt="icon"
    className="TabItem-icon-0-1-5"
    src="iconUrlTest"
  />
  <span
    className="TabItem-name-0-1-6"
  >
    nameTest
  </span>
</div>
`;

exports[`TabItemContainer display properly is selected is mobile 1`] = `
<div
  className="TabItem-tabItem-0-1-1 TabItem-tabItemMobile-0-1-3 TabItem-selected-0-1-4"
  onClick={[Function]}
>
  <img
    alt="icon"
    className="TabItem-icon-0-1-5"
    src="iconUrlTest"
  />
  <span
    className="TabItem-name-0-1-6"
  >
    nameTest
  </span>
</div>
`;

exports[`TabItemContainer display properly is selected is not mobile 1`] = `
<div
  className="TabItem-tabItem-0-1-1 TabItem-tabItemDestop-0-1-2 TabItem-selected-0-1-4"
  onClick={[Function]}
>
  <img
    alt="icon"
    className="TabItem-icon-0-1-5"
    src="iconUrlTest"
  />
  <span
    className="TabItem-name-0-1-6"
  >
    nameTest
  </span>
</div>
`;
